### PR TITLE
Fixed task hang up in shellInternal due to unread error pipe.

### PR DESCRIPTION
**Pull request by grigorye**
This should address https://github.com/spotify/XCRemoteCache/issues/229.

### CODE REVIEW

File: Sources/XCRemoteCache/Shell/Shell.swift [hunk 1]
```diff
@@ -84,13 +84,13 @@ private func shellInternal(_ cmd: String, args: [String] = [], stdout: PipeLike?
         task.currentDirectoryPath = dir
     }
     task.launch()
+    let errorData = (stderr == nil) ? nil : errorHandle.fileHandleForReading.readDataToEndOfFile()
```
Comment (polac24) [inline Sources/XCRemoteCache/Shell/Shell.swift:87]: Shouldn't that leave the `errorData` nil, if custom `stderr` is passed and we cannot inspect it? Only if `stderr` is `nil`, we use our own pipe and then can capture errors.
 
```suggestion
    let errorData = (stderr != nil) ? nil : errorHandle.fileHandleForReading.readDataToEndOfFile()
```

Comment (grigorye) [reply Sources/XCRemoteCache/Shell/Shell.swift]: Yeah, true - sorry, I did not check it.

**Review by polac24 — APPROVED** [review]
Looks good! If you want to land it, make it a PR and we can merge it.

**Comment by grigorye** [comment]
> Looks good! If you want to land it, make it a PR and we can merge it.

Thanks, yeah, fixing this thing may help avoid encountering the hangup in further iterations, if any.

**Comment by polac24** [comment]
thanks!

**Pull request merged**